lib/delayed/performable_method.rb

```diff
@@ -23,7 +24,13 @@ def display_name
     end
 
     def perform
-      object.send(method_name, *args) if object
+      return unless object
+
+      if kwargs.nil? || (RUBY_VERSION < '2.7' && kwargs.empty?)
```

Comment:
There's [a proposed PR](https://github.com/collectiveidea/delayed_job/pull/1130) in the upstream repo, but for this PR I took a very different strategy, because we want to both maintain Ruby 2.6 support and support the Ruby 3 case where both positional and keyword arguments are present. Holding onto `kwargs` separately resolves edge cases around argument auto-merging behaviors, but then on Ruby 2.6 we need to fall back in the `**{}` case (since on 2.6 and below that causes `{}` to get passed as a positional argument).

Reply:
But I think that this should resolve https://github.com/collectiveidea/delayed_job/issues/1134 and might be worth upstreaming, if someone gets a chance.